src/py42/clients/settings/device_settings.py

```diff
@@ -29,12 +29,20 @@ def __init__(self, device_dict, org_settings):
         self.data[u"settings"] = {
             u"serviceBackupConfig": self.data[u"serviceBackupConfig"]
         }
-        self.backup_sets = [
-            BackupSet(self, self.changes, backup_set)
-            for backup_set in self.data[u"serviceBackupConfig"][u"backupConfig"][
-                u"backupSets"
-            ]
-        ]
+        bs = self.data[u"serviceBackupConfig"][u"backupConfig"][u"backupSets"]
+        self.backup_sets = self._extract_backup_sets(bs)
+
+    def _extract_backup_sets(self, backup_sets):
+        if isinstance(backup_sets, dict):  # number of sets are locked
+            backup_sets = backup_sets["backupSet"]
+            if isinstance(backup_sets, dict):  # there's only one set configured
+                return [BackupSet(self, backup_sets)]
+            if isinstance(backup_sets, list):
```

Comment:
This `if` should be an `elif`.

Because in case this implementation changes to not just be `returns` statements and because you are using `else`.